### ISSUE TITLE
A Task Generator that should process only 1 message in parallel, does not work sequential

### ISSUE THREAD

**Issue by MichaelRoeder** [hobbit-project/platform#85]
## Problem

When creating a task generator and setting the maximum number of messages processed in parallel to 1, it is still possible that messages are not processed sequentially.
```
Generated task 0 at 1350203764750 - 1502186858612 of type 2
Generating task 1
Generated task 1 at 1350203787426 - 1502186858625 of type 2
Generating task 2
Generated task 2 at 1350203790552 - 1502186859693 of type 5
Generating task 3
Generated task 3 at 1350203793300 - 1502186863462 of type 2
Generating task 4
Generated task 4 at 1350203794173 - 1502186863463 of type 6
Generating task 5
Generating task 6
Generated task 6 at 1350203804953 - 1502186865187 of type 3
Generating task 7
Generated task 5 at 1350203802214 - 1502186864216 of type 5
```
It can be seen that a task 5 is started to be created but it is finished after task 6 and 7.

**Comment by MichaelRoeder** [comment]
Created a JUnit test to reproduce the problem. (https://github.com/hobbit-project/core/blob/develop/src/test/java/org/hobbit/core/components/TaskGeneratorSequencingTest.java)

**Comment by MichaelRoeder** [comment]
@mirkospasic please check whether the problem is solved when using the latest version `1.0.6-SNAPSHOT` of the core library.

**Comment by mirkospasic** [comment]
Checked... Everything seems to be OK.
